src/CustomDatePickerIOS.js

```diff
@@ -67,6 +67,11 @@ export default class CustomDatePickerIOS extends React.PureComponent {
         date: nextProps.date
       });
     }
+    if(nextProps.maximumDate && (!this.state.date || this.state.date > nextProps.maximumDate)){
```

Comment:
I'm probably missing something: if `state.date` is null/undefined, why should we set it as the `maximumDate`?